corehq/apps/users/management/commands/grant_web_apps_to_default_mobile_worker_role.py

```diff
@@ -0,0 +1,22 @@
+from django.core.management.base import BaseCommand
+
+from corehq.apps.users.analytics import get_role_user_count
+from corehq.apps.users.models_role import Permission, UserRole
+
+
+class Command(BaseCommand):
+    help = "Add access_web_apps permission to default mobile worker roles"
+
+    def handle(self, **options):
+        permission, created = Permission.objects.get_or_create(value='access_web_apps')
+        roles_to_update = UserRole.objects.filter(is_commcare_user_default=True)
+        for role in roles_to_update:
+            permissions = role.permissions
+            if not permissions.access_web_apps:
+                count = get_role_user_count(role.domain, role.couch_id, web_users_only=True)
+                if count == 0:
+                    # Only update if role is limited to mobile workers, who already have access to web apps
+                    if not permissions.web_apps_list:
```

Comment:
For partial access to web apps, `web_apps_list` will be populated, and `access_web_apps` will be false, so it should stay false.

Reply:
Could you use `access_any_web_apps` a few lines above instead of `access_web_apps` to include this check there?  I don't think it needs to happen after the user count thingy.

Reply:
I'm scratching my head hard after not looking at this for three weeks, but yeah, I think you're right, then we'll skip the user count for any roles that are using app-specific permissions, which is a potential performance gain.

Added 8b7383e18c7 - `access_any_web_apps` is a function on a user, not the role itself, so I just moved the existing check.